SimPEG/Regularization.py

```diff
@@ -410,661 +588,807 @@ def eval2Deriv(self, m, v=None):
+    spatial derivative in the provided orientation
 
+    **Optional Inputs**
 
-    @Utils.timeIt
-    def eval(self, m):
-        return self._evalSmall(m) + self._evalSmooth(m) + self._evalSmooth2(m)
+    :param BaseMesh mesh: SimPEG mesh
+    :param int nP: number of parameters
+    :param IdentityMap mapping: regularization mapping, takes the model from model space to the space you want to regularize in
+    :param numpy.ndarray mref: reference model
+    :param numpy.ndarray indActive: active cell indices for reducing the size of differential operators in the definition of a regularization mesh
+    :param numpy.ndarray cell_weights: cell weights
+    :param bool mrefInSmooth: include the reference model in the smoothness computation? (eg. look at Deriv of m (False) or Deriv of (m-mref) (True))
+    :param numpy.ndarray cell_weights: vector of cell weights (applied in all terms)
+    """
 
-    @Utils.timeIt
-    def evalDeriv(self, m):
-        """
-        The regularization is:
+    mrefInSmooth = properties.Bool(
+        "include mref in the smoothness calculation?", default=False
+    )
 
-        .. math::
+    def __init__(
+        self, mesh, orientation='x', **kwargs
+    ):
 
-            R(m) = \\frac{1}{2}\mathbf{(m-m_\\text{ref})^\\top W^\\top W(m-m_\\text{ref})}
+        self.orientation = orientation
 
-        So the derivative is straight forward:
+        assert orientation in ['x', 'y', 'z'], (
+                "Orientation must be 'x', 'y' or 'z'"
+            )
 
-        .. math::
+        if self.orientation == 'y':
+            assert mesh.dim > 1, (
+                "Mesh must have at least 2 dimensions to regularize along the "
+                "y-direction"
+            )
 
-            R(m) = \mathbf{W^\\top W (m-m_\\text{ref})}
+        elif self.orientation == 'z':
+            assert mesh.dim > 2, (
+                "Mesh must have at least 3 dimensions to regularize along the "
+                "z-direction"
+            )
 
-        """
-        return self._evalSmallDeriv(m) + self._evalSmoothDeriv(m) + self._evalSmoothDeriv2(m)
+        super(SmoothDeriv, self).__init__(
+            mesh=mesh, **kwargs
+        )
 
-    def eval2Deriv(self, m, v=None):
+        if self.mrefInSmooth is False:
+            self.mref = Utils.Zero()
+
+    @property
+    def _multiplier_pair(self):
+        return 'alpha_{orientation}'.format(orientation=self.orientation)
+
+    @property
+    def W(self):
         """
-        The regularization is:
+        Weighting matrix that constructs the first spatial derivative stencil
+        in the specified orientation
+        """
+        W = getattr(
+            self.regmesh,
+            "cellDiff{orientation}".format(
+                orientation=self.orientation
+            )
+        )
+        if self.cell_weights is not None:
+            Ave = getattr(self.regmesh, 'aveCC2F{}'.format(self.orientation))
+            W = (
```

Comment:
This has been addressed in #592. The diff is a bit confusing as I created a `Small` (with vols) and a `SimpleSmall` (without vols)